src/mlpack/methods/sparse_autoencoder/maximal_inputs.hpp

```diff
@@ -0,0 +1,48 @@
+#ifndef __MLPACK_METHODS_NN_MAXIMAL_INPUTS_HPP
+#define __MLPACK_METHODS_NN_MAXIMAL_INPUTS_HPP
+
+#include <mlpack/core.hpp>
+
+namespace mlpack {
+namespace nn {
+
+
+/**
+ * Maximize the hidden units of the parameters
```

Comment:
Do you think you could add a bit more documentation on what the function does and what it's for?  A link to the UFLDL page would probably be pretty useful here.  (If you don't want to do this, I can do it after the merge.)